core/src/Cuda/Kokkos_Cuda_Parallel_Team.hpp

```diff
@@ -47,6 +47,14 @@ extern bool show_warnings() noexcept;
 
 namespace Impl {
 
+// Cuda Teams use (team_size + 2)*sizeof(double) shared memory for team
+// reductions. They also use one int64_t in static shared memory for a shared
+// ID. Furthermore, they use additional scratch memory in some reduction
+// scenarios, which depend on the size of the value_type and is NOT captured
+// here.
+constexpr size_t cuda_team_max_reserved_shared_mem =
+    (1024 + 2) * sizeof(double) + sizeof(int64_t);
```

Comment:
Why declare it at namespace scope?   Keep the scope minimal, move it to `TeamPolicyInternal<Cuda>:: scratch_size_max()`